ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -0,0 +1,305 @@
+
+        for key, value in engine.state.metrics.items():
+            writer.add_scalar(tag=key,
+                              scalar_value=value,
+                              global_step=global_step)
+
+    def _plot_state_keys(self, engine, writer, state_keys, mode='epoch'):
+        """
+        Plots engine.state attributes on epoch or iteration.
+
+        Args:
+            engine (ignite.Engine): training engine or evaluator engine
+            state_keys (list): list of keys of engine.state attributes
+            mode (str): 'epoch' or 'iteration'
+        """
+
+        global_step = self.engine.state.epoch if mode == 'epoch' else self.engine.state.iteration
+        for key in state_keys:
+
+            if hasattr(engine.state, key):
+                state = getattr(engine.state, key)
+            else:
+                raise KeyError('{} not an attribute of engine.state'.format(key))
+
+            if isinstance(state, (int, float)):
+                writer.add_scalar(tag=key,
+                                  scalar_value=state,
+                                  global_step=global_step)
+            else:
+                raise ValueError('engine.state.{} should be of instance int or float.'.format(key))
+
+    def _plot_output(self, engine, writer, output_transform):
+        global_step = self.engine.state.iteration
+
+        output = output_transform(engine.state.output)
+
+        if not isinstance(output, dict):
+            raise ValueError("output_transform must transform engine.state.output into a dictionary.")
+
+        for key, value in output.items():
+            if isinstance(value, (int, float)):
+                writer.add_scalar(tag=key,
+                                  scalar_value=value,
+                                  global_step=global_step)
+            else:
+                raise ValueError('engine.state.output[\'{}\'] should be of instance int or float.'.format(key))
+
+    def _update_iteration(self, engine, mode, use_metrics, use_output, output_transform, state_keys, learning):
+
+        writer = self.writer[learning]
+
+        if mode == 'iteration':
+            if use_metrics:
+                self._plot_metrics(engine, writer, mode='iteration')
+            if state_keys:
+                self._plot_state_keys(engine, writer, state_keys, mode='iteration')
+
+        if use_output:
+            self._plot_output(engine, writer, output_transform)
+
+    def _update_epoch(self, engine, mode, use_metrics, state_keys, model, histogram_freq, write_grads, learning):
+
+        writer = self.writer[learning]
+
+        if mode == 'epoch':
+            if use_metrics:
+                self._plot_metrics(engine, writer, mode='epoch')
+
+            if state_keys:
+                self._plot_state_keys(engine, writer, state_keys, mode='epoch')
+
+        if histogram_freq > 0 and self.engine.state.epoch % histogram_freq == 0:
```

Comment:
I expect histogram_freq to be 1/histogram_interval. Then the following should be correct from my understanding:
```suggestion
        if histogram_freq > 0 and self.engine.state.epoch % 1./histogram_freq == 0:
```